src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -1791,6 +1791,19 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         tmp = builder->CreateFSub(exp, one);
     }
 
+    void generate_Input0() {
+        std::string input;
+        std::cin >> input;
+        tmp = builder->CreateGlobalStringPtr(input);
+    }
+
+    void generate_Input1(ASR::expr_t* m_arg) {
+        std::string input;
+        std::cout << ASR::down_cast<ASR::StringConstant_t>(m_arg)->m_s;
+        std::cin >> input;
```

Comment:
@kmr-srbh this reads input at `compile-time`, but rather you need the input to be read at runtime. 

For example, I think your approach would fail when just compiling the source to a binary file (called `tmp` here) as follows: 
```console
lpython$ ./src/bin/lpython ./examples/example.py -o tmp
```
I think the compiler would wait for the input but the input would actually be provided when the `tmp` file is run. That is:
```console
./tmp
>>> Now provide your input here
```

Reply:
@Shaikh-Ubaid You are right. How do we go about working a fix? Please guide me.

Reply:
>How do we go about working a fix?

We handle File I/O by calling into the C runtime library (`lfortran_intrinsics.c`). So for taking input from the terminal (i.e. standard I/O), I think we might need to do the same. LFortran already supports taking input from standard I/O. You can check it out for a reference. You can also checkout https://github.com/lfortran/lfortran/pull/2025.

Reply:
Great resource! I have a question. How will we link it to the `input()` intrinsic here?

Reply:
>How will we link it to the input() intrinsic here?

We use `visit_FileRead()` which reads from `stdin` when the `m_unit` is not provided. I shared more about implementing it here https://github.com/lcompilers/lpython/pull/2601#discussion_r1527257564. You can also check how things work in LFortran for specific details.